packages/neovim/src/utils/getNvimFromEnv.ts

```diff
@@ -0,0 +1,202 @@
+   * This will be empty if all versions match the minimum version or if no minimum version was specified.
+   * This will not be sorted (it will be in the order of the paths in the `PATH` environment variable).
+   */
+  readonly unmatchedVersions: ReadonlyArray<NvimVersion>;
+  /**
+   * A list of errors that occurred while trying to get the nvim versions.
+   * This will be empty if no errors occurred.
+   * This will not be sorted (it will be in the order of the paths in the `PATH` environment variable).
+   * Unmatched versions are not treated as errors.
+   */
+  readonly errors: ReadonlyArray<GetNvimFromEnvError>;
+};
+
+const versionRegex = /^(\d+)\.(\d+)\.(\d+)(?:-(.+))?$/;
+const nvimVersionRegex = /^NVIM\s+v(.+)$/m;
+const buildTypeRegex = /^Build\s+type:\s+(.+)$/m;
+const luaJitVersionRegex = /^LuaJIT\s+(.+)$/m;
+const windows = process.platform === 'win32';
+
+export function parseVersion(version: string): (number | string)[] | null {
+  if (typeof version !== 'string') {
+    throw new Error('Invalid version format: not a string');
+  }
+
+  const match = version.match(versionRegex);
+  if (match === null) {
+    return null;
+  }
+
+  const [, major, minor, patch, prerelease] = match;
+  const majorNumber = Number(major);
+  if (Number.isNaN(majorNumber)) {
+    throw new Error('Invalid version format: major is not a number');
+  }
+
+  const minorNumber = Number(minor);
+  if (Number.isNaN(minorNumber)) {
+    throw new Error('Invalid version format: minor is not a number');
+  }
+
+  const patchNumber = Number(patch);
+  if (Number.isNaN(patchNumber)) {
+    throw new Error('Invalid version format: patch is not a number');
+  }
+
+  const versionParts: Array<number | string> = [
+    majorNumber,
+    minorNumber,
+    patchNumber,
+  ];
+  if (prerelease !== undefined) {
+    versionParts.push(prerelease);
+  } else {
+    versionParts.push('zzz');
+  }
+  return versionParts;
+}
+
+/**
+ * Compare two versions.
+ * @param a - The first version to compare.
+ * @param b - The second version to compare.
+ * @returns -1 if a < b, 0 if a == b, 1 if a > b.
+ * @throws {Error} If the versions are not valid.
+ *
+ * Format could be:
+ * - 0.9.1
+ * - 0.10.0-dev-658+g06694203e-Homebrew
+ */
+export function compareVersions(a: string, b: string): number {
+  const versionA = parseVersion(a);
+  const versionB = parseVersion(b);
+  const length = Math.min(versionA.length, versionB.length);
+
+  for (let i = 0; i < length; i = i + 1) {
+    const partA = versionA[i];
+    const partB = versionB[i];
+    if (partA < partB) {
+      return -1;
+    }
+    if (partA > partB) {
+      return 1;
+    }
+  }
+
+  if (versionB.length > versionA.length) {
+    return -1;
+  }
+
+  return 0;
+}
+
+/**
+ * Get the highest matching nvim version from the environment.
+ */
+export function getNvimFromEnv(
+  opt: GetNvimFromEnvOptions = {}
+): Readonly<GetNvimFromEnvResult> {
+  const paths = process.env.PATH.split(delimiter);
```

Comment:
$PATH is likely not sufficient, especially for applications such as https://github.com/vscode-neovim/vscode-neovim which are started in GUI contexts and often (macOS, at least) the user's terminal $PATH is much different than whatever GUI apps are started with.

Thus we also need to search "common locations". These are the ones I know of:

```
/usr/local/bin/
/usr/bin
/opt/homebrew/bin
/home/linuxbrew/.linuxbrew/bin
$HOME/.linuxbrew/bin
$HOME/bin
```